src/components/button-toggle/button-toggle.ts

```diff
@@ -0,0 +1,327 @@
+      }
+    }
+  }
+
+  /** Dispatch change event with current selection and group value. */
+  private _emitChangeEvent(): void {
+    let event = new MdButtonToggleChange();
+    event.source = this._selected;
+    event.value = this._value;
+    this._change.emit(event);
+  }
+}
+
+/** Multiple selection button-toggle group. */
+@Directive({
+  selector: 'md-button-toggle-group[multiple]',
+})
+export class MdButtonToggleGroupMultiple {
+  /** Disables all toggles in the group. */
+  private _disabled: boolean = null;
+
+  /** Child button toggles. */
+  @ContentChildren(forwardRef(() => MdButtonToggle))
+  private _buttonToggles: QueryList<MdButtonToggle> = null;
+
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+
+  set disabled(value) {
+    this._disabled = (value != null && value !== false) ? true : null;
+  }
+}
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-button-toggle',
+  templateUrl: 'button-toggle.html',
+  styleUrls: ['button-toggle.css'],
+  encapsulation: ViewEncapsulation.None
+})
+export class MdButtonToggle implements OnInit {
+  /** Whether or not this button toggle is checked. */
+  private _checked: boolean = false;
+
+  /** Type of the button toggle. Either 'radio' or 'checkbox'. */
+  private _type: string = null;
+
+  /** The unique ID for this button toggle. */
+  @HostBinding('id')
+  @Input()
+  id: string;
+
+  /** HTML's 'name' attribute used to group radios for unique selection. */
+  @Input()
+  name: string;
+
+  /** Whether or not this button toggle is disabled. */
+  private _disabled: boolean = null;
+
+  /** Value assigned to this button toggle. */
+  private _value: any = null;
+
+  /** The parent button toggle group (exclusive selection). Optional. */
+  buttonToggleGroup: MdButtonToggleGroup;
+
+  /** The parent button toggle group (multiple selection). Optional. */
+  buttonToggleGroupMultiple: MdButtonToggleGroupMultiple;
+
+  /** Event emitted when the group value changes. */
+  @Output()
+  change: EventEmitter<MdButtonToggleChange> = new EventEmitter<MdButtonToggleChange>();
+
+  constructor(@Optional() toggleGroup: MdButtonToggleGroup,
+              @Optional() toggleGroupMultiple: MdButtonToggleGroupMultiple,
+              public buttonToggleDispatcher: MdUniqueSelectionDispatcher) {
+    this.buttonToggleGroup = toggleGroup;
+
+    this.buttonToggleGroupMultiple = toggleGroupMultiple;
+
+    buttonToggleDispatcher.listen((id: string, name: string) => {
+      if (id != this.id && name == this.name) {
+        this.checked = false;
+      }
+    });
+  }
+
+  ngOnInit() {
+    if (this.id == null) {
+      this.id = `md-button-toggle-${_uniqueIdCounter++}`;
+    }
+
+    if (this.buttonToggleGroup && this._value == this.buttonToggleGroup.value) {
+      this._checked = true;
+    }
+
+    // Even if there is no group at all, treat the button toggle as a checkbox so it can be
+    // toggled on or off.
+    if (this.buttonToggleGroup) {
```

Comment:
You should be able to do this check in the constructor since you'll know at that point which type this toggle is.